### DISCUSSION TITLE
Is PHPStan supposed to be supported?

### DISCUSSION THREAD

**Discussion by mnapoli** [pestphp/pest#670] [Q&A] [answered]
I just installed Pest and I'm getting loads of PHPStan errors.

Going through the past issues, it seems like PHPStan is supposed to be supported, but I'd like to make sure?

Should I install anything specific so that PHPStan understands the Pest API and its tests?

For example `$this->assertSame(` or ` mock(SsmClient::class)->expects` raises errors in phpstan.

**Comment by nunomaduro** [comment, marked as answer]
PHPStan does not understand Pest's `$this`. However, you can use Pest's expectation API, or PHPUnit's Assert class directly if you really want to use PHPStan in your test suite.

**Reply by mnapoli** [reply]
But `mock(SsmClient::class)->expects` is Pest's expectation API? Or is it not supported either?

**Reply by girardotcl** [reply]
I opened an issue, i'm very sorry if i made duplication, but :
`test()` replacement, is it a solid and sustainable solution from this package point of view ?
https://github.com/pestphp/pest/issues/1104#issue-2158461319

**Comment by rburgst** [comment]
see https://github.com/larastan/larastan/issues/706#issuecomment-2622755854
